specifyweb/frontend/js_src/lib/components/AppResources/TabDefinitions.tsx

```diff
@@ -96,6 +97,24 @@ const generateEditor = (xmlSpec: (() => BaseSpec<SimpleXmlNode>) | undefined) =>
       [stateRestored]
     );
     const isReadOnly = React.useContext(ReadOnlyContext);
+
+    const displayWarningPanel = React.useMemo(
+      () => () => {
+        f.maybe(codeMirrorRef.current?.view, openLintPanel);
+      },
+      [codeMirrorRef]
+    );
+
+    React.useEffect(() => {
+      const timeoutId = setTimeout(() => {
```

Comment:
is setTimeout necessary? try to avoid it whenever possible as setTimeout is a hacky solution
i.e could you instead display the warnings panel when the first lint error occurs? (which may happen on page load, or later caused by user edit)

Reply:
@maxxxxxdlp,  I change the logic in my new commit but I cannot figure out why now the diagnostic is always empty even if the `warnings` array shows the proper warnings...